test/dns-test.js

```diff
@@ -0,0 +1,396 @@
+
+      // Handshake zone is not authoritative for ICANN based name
+      assert.equal(res.aa, false);
+    });
+
+    it('should return no answer for HNS name pre-update', async () => {
+      const name = 'foo.';
+
+      const res = await nstub.lookup(name, types.A);
+      assert.equal(res.answer.length, 0);
+    });
+
+    // record to save in the authenticated tree
+    for (const [hstype, item] of Object.entries(json)) {
+      const type = types[item.type];
+      const resource = Resource.fromJSON(item.resource);
+
+      // args: size, height, network
+      const name = rules.grindName(5, 1, network);
+
+      it(`should update ${name}`, async () => {
+        await wallet.client.post(`/wallet/${wallet.id}/open`, {
+          name: name
+        });
+
+        await mineBlocks(treeInterval + 1, coinbase);
+
+        await wallet.client.post(`/wallet/${wallet.id}/bid`, {
+          name: name,
+          bid: 1000,
+          lockup: 2000
+        });
+
+        await mineBlocks(biddingPeriod + 1, coinbase);
+
+        await wallet.client.post(`/wallet/${wallet.id}/reveal`, {
+          name: name
+        });
+
+        await mineBlocks(revealPeriod + 1, coinbase);
+
+        await wallet.client.post(`/wallet/${wallet.id}/update`, {
+          name: name,
+          data: resource.toJSON()
+        });
+
+        // mine a block
+        await mineBlocks(1, coinbase);
+
+        const rinfo = await nclient.execute('getnameresource', [name]);
+
+        const json = resource.getJSON(name);
+        assert.deepEqual(rinfo, json);
+
+        const info = await nclient.execute('getnameinfo', [name]);
+        const data = Buffer.from(info.info.data, 'hex');
+
+        const returned = Resource.decode(data);
+        assert.deepEqual(returned.getJSON(name), json);
+
+        // this commits the tree state to disk
+        // updates served by the dns server
+        await mineBlocks(treeInterval, coinbase);
+      });
+
+      it(`should return authenticated ${hstype} record`, async () => {
+        // Certain types of queries require the
+        // name to be formatted with additional data
+        const query = buildQuery(name, resource, type);
+
+        const response = await nstub.lookup(query, type);
+        const res = response.toJSON();
+        const dns = resource.toDNS(util.fqdn(name), type).toJSON();
+
+        // query for the zone signing key
+        const dnskey = await nstub.lookup('.', types.DNSKEY);
+
+        // parse the zone signing key and
+        // key signing key out of the response
+        const {zsk, ksk} = getSigningKeys(dnskey);
+
+        assert(zsk instanceof wire.Record);
+        assert(ksk instanceof wire.Record);
+
+        // validate the signature on the ZSK
+        verifyDNSSEC(dnskey, ksk, types.DNSKEY, '.');
+        // validate the signature over the rrsets
+        verifyDNSSEC(response, zsk, type, query);
+
+        // NOTE: the signatures are not canonical
```

Comment:
does this mean `verifyDNSSEC` manages to verify non-canonical signatures? If yes, is it possible to convert to canonical or otherwise verify them instead of skipping them altogether (like nullSig is doing now I presume).

Reply:
`verifyDNSSEC` does work on non-canonical signatures, the problem was that a `deepEqual` did not work because the signatures were not canonical when using the native backend that wraps openssl.

Reply:
OK.